packages/Input/src/Input.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect } from 'react';
+import cx from 'classnames';
+
+import './input.css';
+
+const KeyCodes = {
+  Comma: 44,
+  Minus: 45,
+  Nine: 57,
+  Period: 46,
+  Plus: 43,
+  Zero: 48,
+};
+
+export type InputSize = 'small' | 'medium';
+
+export type InputType = 'email' | 'text' | 'password' | 'number';
+
+export interface InputProps extends React.ComponentPropsWithRef<'input'> {
+  // Add class names to the surrounding DOM container.
+  className?: string;
+  // Form.ValidatedField state. "Error" if has error, otherwise null.
+  state?: string;
+  // Specifies the type to display
+  type?: InputType;
+  // Input size.
+  inputSize?: InputSize;
```

Comment:
Pour garder une cohérence dans la nomenclature des props entre chaque composant peut tu le renommer `size: InputSize;`

Reply:
Je n'ai pas pu la nommer "size" car il y a un conflit avec l'attribut existant de input

`Interface 'InputProps' incorrectly extends interface 'Pick<DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, "key" | keyof InputHTMLAttributes<...>> & { ...; }'.
  Type 'InputProps' is not assignable to type 'Pick<DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, "key" | keyof InputHTMLAttributes<...>>'.
    Types of property 'size' are incompatible.
      Type 'InputSize | undefined' is not assignable to type 'number | undefined'.
        Type '"small"' is not assignable to type 'number | undefined'.ts(2430)`

Reply:
Effectivement ça entre en conflit avec la `size` native des inputs, ce que je te propose c'est de remplacer cette propriété par:  `isCompact?: boolean;` qu'en dis tu?